python/ptest/cases/gomspace_checkout_case.py

```diff
@@ -0,0 +1,152 @@
+        print("cursys is: " + str(cursys) + " mA")
+
+        curout = [int(read_state(self, "gomspace.curout.output" + str(i)))
+                  for i in range(1, 7)]
+        for n in range(0, len(curout)):
+            print("curout" + str(n) + " is: " + str(curout[n]) + " mA")
+
+        output = [self.str_to_bool(read_state(self, "gomspace.output.output" + str(i)))
+                  for i in range(1, 7)]
+
+        for n in range(0, len(output)):
+            out_n = output[n]
+            # checked umbilical for which outputs should be 0 and 1:
+            # 1-5 are all 5V, 6 is 3.3V
+            if out_n is False:
+                print("Output-" + str(n) + " is not on")
+
+        wdt_i2c_time_left = int(read_state(self, "gomspace.wdt_i2c_time_left"))
+        if wdt_i2c_time_left < 99:
+            print("wdt_i2c_time_left is less than 99 seconds at: " +
+                  str(wdt_i2c_time_left))
+
+        counter_wdt_i2c = read_state(self, "gomspace.counter_wdt_i2c")
+        print("counter_wdt_i2c is: " + str(counter_wdt_i2c))
+        counter_boot = read_state(self, "gomspace.counter_boot")
+        print("counter_wdt_i2c is: " + str(counter_boot))
+
+        temp = [int(read_state(self, "gomspace.temp.output" + str(i)))
+                for i in range(1, 5)]
+        for n in range(0, len(temp)):
+            temp_n = temp[n]
+            if temp_n < 20 or out_n > 25:
+                print("Temp-" + str(n) +
+                      " is out of room temperature range [20, 25] degC at: " + str(temp_n))
+
+        bootcause = read_state(self, "gomspace.bootcause")
+        print("bootcause is: " + str(bootcause))
+        battmode = read_state(self, "gomspace.battmode")
+        print("battmode is: " + str(battmode))
+        pptmode = read_state(self, "gomspace.pptmode")
+        print("pptmode is: " + str(pptmode))
+
+        # writable fields
+        power_cycle_output_cmd = [self.str_to_bool(read_state(self, "gomspace.power_cycle_output" + str(i) + "_cmd"))
+                                  for i in range(1, 7)]
+        cycle_no_init = int(read_state(self, "pan.cycle_no"))
+        cycle_no = cycle_no_init
+
+        # power cycling
+        # start power cycle
+        power_cycle_output_cmd = [self.str_to_bool(write_state(self, "gomspace.power_cycle_output"
+                                                               + str(i) + "_cmd", "true"))
+                                  for i in range(1, 7)]
+        # wait for outputs to be off
+        while (not all(out == False for out in output)) and cycle_no - cycle_no_init < 600:
+            output = [self.str_to_bool(read_state(self, "gomspace.output.output" + str(i)))
+                      for i in range(1, 7)]
+            simulation.flight_controller.write_state(
+                self, "pan.cycle_no", cycle_no + 1)
+            cycle_no = int(read_state(self, "pan.cycle_no"))
+        # wait for outputs to turn on again
```

Comment:
After this while loop, you should probably print an error if the outputs still haven't turned off after 1 minute